### DISCUSSION TITLE
Some questions about recognition

### DISCUSSION THREAD

**Discussion by phirestalker** [nextcloud/recognize#1039] [General]
I was wondering why the 120 faces to start clustering? On a related thread, if there are thousands of images in queue for face recognition, would it be possible to wait for clustering until all of that is done, so there aren't any faces left after it's done? Or, is it that each run needs exactly 120 faces, no more, no less?
I would also like to know if the object recognition only searches for one "main" thing in the photos, or a set limit of objects in one image. Is there a size limit for the objects, like nothing too small? Does the object recognition use thumbnails as well? If so, I guess that would limit how many objects it could recognize.
Does the model used for object recognition ever get updated to add new objects, or did the people who trained it just leave it as is?

EDIT:
One more I forgot. The tags that are set for the files in Nextcloud, are they also saved to the EXIF metadata, or equivalent?

**Comment by marcelklehr** [comment]
Hi @phirestalker :wave: 

> I was wondering why the 120 faces to start clustering?

120 faces because the clustering algorithm works better the more data we have. We do count already clustered faces into that threshold, so when you already have a lot of clustered faces, it's likely you won't run into this limit.

> is it that each run needs exactly 120 faces, no more, no less?

An automatically scheduled clustering run tries to aim for 10000 faces to cluster in one go. It won't cluster anything unless it has more than 120 faces available from new, unclustered + already clustered faces.

> I would also like to know if the object recognition only searches for one "main" thing in the photos, or a set limit of objects in one image.

The object recognition usually looks for the main thing in the image. So if you have a big elephant next to a small staircase, it might only get the elephant. I would not count on the current model recognizing more than one or two objects in a scene. It wasn't built for that.

> Does the model used for object recognition ever get updated to add new objects, or did the people who trained it just leave it as is?

The model hasn't been updated so far. We might do that at some point.

> The tags that are set for the files in Nextcloud, are they also saved to the EXIF metadata, or equivalent?

No, recognize as a whole currently does not deal with image metadata at all.

**Reply by shaun-ba** [reply]
.

**Reply by marcelklehr** [reply]
Yes
